### ISSUE TITLE
ValueError: Expected 2D array, got 1D array instead

### ISSUE THREAD

**Issue by hjzhang1018** [deepchem/deepchem#2137]
Hi, I'm using the Random Forest model to study my dataset. Following is the code I used. After the `model.fit(train_dataset)`, however it shows an error: "ValueError: Expected 2D array, got 1D array instead:
array=[1. 0. 0. ... 0. 0. 0.].Reshape your data either using array.reshape(-1, 1) if your data has a single feature or array.reshape(1, -1) if it contains a single sample."

My data is single task, the "label" column contains only 0s and 1s, dtype is int. I've searched similar issues and the suggested solution is to reshape the value according to the hint. But I'm confused if I should use `model.fit(train_dataset.X, train_dataset.y.reshape(-1,1))` or other formats to only reshape the "label" array? Any advice is greatly appreciated!

```
dataset_file= "data.csv"
dataset = pd.read_csv("data.csv")
featurizer = dc.feat.CircularFingerprint(size=1024)
loader = dc.data.CSVLoader(
      tasks=["label"], feature_field="Smiles",
      featurizer=featurizer)
dataset = loader.featurize(dataset_file)
splitter = dc.splits.ScaffoldSplitter()
train_dataset, valid_dataset, test_dataset = splitter.train_valid_test_split(dataset)
transformers = [dc.trans.NormalizationTransformer(transform_X=True, dataset=train_dataset)]
for dataset in [train_dataset, valid_dataset, test_dataset]:
   for transformer in transformers:
       dataset = transformer.transform(dataset)
from sklearn.ensemble import RandomForestClassifier
sklearn_model = RandomForestClassifier
model = dc.models.SklearnModel(sklearn_model)
model.fit(train_dataset)

```

**Comment by rbharath** [comment]
Glad to help out! As a first comment, I think this code snippet has a couple bugs. The lines

```
for dataset in [train_dataset, valid_dataset, test_dataset]:
   for transformer in transformers:
       dataset = transformer.transform(dataset)
```

only overwrite the `dataset` variable and don't actually replace the train/valid/test datasets. For that you'll want

```
transformer = dc.trans.NormalizationTransformer(transform_X=True, dataset=train_dataset)
train_dataset = transformer.transform(train_dataset)
valid_dataset = transformer.transform(valid_dataset)
test_dataset = transformer.transform(test_dataset)
```

The second bug is that `SklearnModel` requires an instantiated sklearn model as input. So you'll want

```
sklearn_model = RandomForestClassifier()
model = dc.models.SklearnModel(sklearn_model)
```

Can you give it a try with these fixes and see if it works for you?

**Comment by hjzhang1018** [comment]
Thank you @rbharath !
I tried but still there's a problem showing that 
"ValueError                                Traceback (most recent call last)
<ipython-input-15-8fb4d4cad5e4> in <module>()
      2 sklearn_model = RandomForestClassifier()
      3 model = dc.models.SklearnModel(sklearn_model)
----> 4 model.fit(train_dataset)

3 frames
/usr/local/lib/python3.6/dist-packages/sklearn/utils/multiclass.py in check_classification_targets(y)
    167     if y_type not in ['binary', 'multiclass', 'multiclass-multioutput',
    168                       'multilabel-indicator', 'multilabel-sequences']:
--> 169         raise ValueError("Unknown label type: %r" % y_type)
    170 
    171 

ValueError: Unknown label type: 'unknown'"

I looked at the data shape, `train_dataset.X.shape` is (6495, 1024) and `train_dataset.y.shape` is (6495, 1). I don't understand why this is not working.

Thank you again!

**Comment by rbharath** [comment]
My guess is that you have a regression dataset instead of a classification dataset. Try a `RandomForestRegressor()` instead and see if that works :)

**Comment by hjzhang1018** [comment]
@rbharath Thank you for your advice! My dataset is a classification dataset though. I tried the same code locally using Pycharm, and no such value error show up. I'm a bit confused why this is happening :)

**Comment by rbharath** [comment]
Would it be possible to give me a short failing coding snippet? If I can reproduce the error on my machine, I'd be glad to take a stab at debugging it!

**Comment by hjzhang1018** [comment]
Yes sure, here is a short coding snippet. Thank you for your help! This works in Pycharm but not in Colab.
```
dataset_file= "data.csv"
dataset = pd.read_csv("data.csv")
featurizer = dc.feat.CircularFingerprint(size=1024)
loader = dc.data.CSVLoader(
      tasks=["label"], feature_field="Smiles",
      featurizer=featurizer)
dataset = loader.featurize(dataset_file)
splitter = dc.splits.ScaffoldSplitter()
train_dataset, valid_dataset, test_dataset = splitter.train_valid_test_split(dataset)
from sklearn.ensemble import RandomForestClassifier
rf_model = RandomForestClassifier(n_estimators=100)
model = dc.models.SklearnModel(rf_model)
model.fit(train_dataset, nb_epoch=10)

roc_auc_metric = Metric(metrics.roc_auc_score)
recall_metric = Metric(metrics.recall_score)
precision_metric = Metric(metrics.precision_score, threshold=0.5)
accuracy_metric = Metric(metrics.accuracy_score)
mcc_metric = Metric(metrics.matthews_corrcoef)
all_metrics = [roc_auc_metric, recall_metric, precision_metric, accuracy_metric, mcc_metric]

model.evaluate(valid_dataset, all_metrics, [])
model.evaluate(test_dataset, all_metrics, [])
```

**Comment by rbharath** [comment]
Great thanks! Could you also append a prefix from "data.csv" if you can that I can use to run the sample code? If the data is private, you can obscure it by swapping out random compounds/numbers so long as the style/format of everything is the same :) and you can reproduce the error with the new file
